src/components/Common/View.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { ComponentType } from "react";
+
+import { useView } from "@/Utils/useView";
+
+export default function View({
+  name,
+  board,
+  list,
+}: {
+  name: string;
+  board: ComponentType<{ setView: (view: string) => void }>;
+  list: ComponentType<{ setView: (view: string) => void }>;
+}) {
+  const [view, setView] = useView(`${name}DefaultView`);
+
+  const views: Record<
+    "board" | "list",
+    ComponentType<{ setView: (view: string) => void }>
```

Comment:
why pass set view as prop? we can always use `useView` again if child needs to change the view. this is not reliable since we'll have to make sure the implemented components always has correct props types which is unnecessary